### PR TITLE
Fix minio distributed setup to properly work on windows

### LINKED ISSUE #5136: Run distributed Minio - Windows
The command in the document seem to have issues.

minio.exe server http://192.168.1.11/C:/data http://192.168.1.12/C:/data ^
                  http://192.168.1.13/C:/data http://192.168.1.14/C:/data ^
                  http://192.168.1.15/C:/data http://192.168.1.16/C:/data ^
                  http://192.168.1.17/C:/data http://192.168.1.18/C:/data

Not sure if there is an issue with escaping. Minio service never start with the above command. Replaced the IP address with my ip address.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
On windows having a preceding "/" will cause problems, if the
command line already has C:/<export-folder/ in it. Final resulting
path on windows might become C:/C:/ this will cause problems
of starting minio server properly in distributed mode on windows.
As a special case make sure to trim the separator.

NOTE: It is also perfectly fine for windows users to have a path
without C:/ since at that point we treat it as relative path
and obtain the full filesystem path as well. Providing C:/
style is necessary to provide paths other than C:/,
such as F:/, D:/ etc.

Another additional benefit here is that this style also
supports providing UNC paths as well.

## Motivation and Context
Fixes #5136

## How Has This Been Tested?
Manually on windows.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/endpoint.go [hunk 1]
```diff
@@ -130,6 +130,24 @@ func NewEndpoint(arg string) (ep Endpoint, e error) {
...
+
+		// NOTE: It is also perfectly fine for windows users to have a path
+		// without C:/ since at that point we treat it as relative path
+		// and obtain the full filesystem path as well. Providing C:/
+		// style is necessary to provide paths other than C:/,
+		// such as F:/, D:/ etc.
+		//
+		// Another additional benefit here is that this style also
+		// supports providing \\host\share support as well.
+		if filepath.VolumeName(u.Path[1:]) != "" {
```
Comment (balamurugana) [inline cmd/endpoint.go]: you would need to `runtime.GOOS == "windows"` as well.  A crazy `/c:` directory is ignored in unixes

**Comment by harshavardhana** [comment]
Can you guys review ?

**Review by balamurugana — DISMISSED** [review]

File: cmd/endpoint.go [hunk 2]
```diff
@@ -130,6 +131,26 @@ func NewEndpoint(arg string) (ep Endpoint, e error) {
 			return ep, fmt.Errorf("empty or root path is not supported in URL endpoint")
 		}
 
+		// On windows having a preceding "/" will cause problems, if the
+		// command line already has C:/<export-folder/ in it. Final resulting
+		// path on windows might become C:/C:/ this will cause problems
+		// of starting minio server properly in distributed mode on windows.
+		// As a special case make sure to trim off trim off the separator.
```
Comment (nitisht) [inline cmd/endpoint.go]: trim off is added twice

Comment (harshavardhana) [reply cmd/endpoint.go]: Good catch will fix

**Comment by harshavardhana** [comment]
Pushed changes after @nitisht review - PTAL @balamurugana @donatello

**Review by krishnasrinivas — APPROVED** [review]
Do you have a VM to test this @harshavardhana

**Comment by harshavardhana** [comment]
> Do you have a VM to test this @harshavardhana

I had on packet.net @krishnasrinivas you can start your own 
[minio.exe.zip](https://github.com/minio/minio/files/1459600/minio.exe.zip)

**Review by balamurugana — APPROVED** [review]
LGTM

**Comment by harshavardhana** [comment]
Coverage cannot be addressed since we disabled coverage on windows out of go tooling bug.  We have a relevant bug open to enable once we have tooling fixed in go perhaps in 1.10

**Pull request merged**